### PR TITLE
Add Autosense annotation for query dsl testing

### PR DESCRIPTION
**Pull request by MaineC**
This adds the autosense annotation to a couple of query dsl docs files and fixes the snippets to work in the tests along the way.

Still needed: Switch from autosense to console, convert remaining query dsl docs.

Relates to #18160 / @nik9000

### CODE REVIEW

File: docs/reference/query-dsl/constant-score-query.asciidoc [hunk 1]
```diff
@@ -7,12 +7,16 @@ filter. Maps to Lucene `ConstantScoreQuery`.
 
 [source,js]
 --------------------------------------------------
+GET /_search
 {
+  "query": {
```
Comment (nik9000) [inline docs/reference/query-dsl/constant-score-query.asciidoc]: I'm not sure if it is ok to confuse indention? I like two spaces and some of the snippets use that but this uses both two and four spaces now.

File: docs/reference/query-dsl/geo-distance-query.asciidoc [hunk 2]
```diff
@@ -15,13 +40,18 @@ distance from a geo point. Assuming the following indexed json:
     }
 }
 --------------------------------------------------
+// CONSOLE
+// TEST[continued]
```
Comment (nik9000) [inline docs/reference/query-dsl/geo-distance-query.asciidoc]: Thanks!

Comment (nik9000) [inline docs/reference/query-dsl/geo-distance-query.asciidoc]: Another option, so you don't have to do `// TEST[continued]` everywhere is to do `// TESTSETUP` but it only works with one snippet right now I think. So if you put the mapping and document into a single snippet you could use that instead. Then each snippet would have those commands run first. I think I'd prefer something like that because the snippets are examples rather than a single story? It is way too early with the annotations for me to have a good idea of what a good style is but that is my instinct at least.

File: docs/reference/query-dsl/constant-score-query.asciidoc (hunk 1 above)
Comment (MaineC) [inline docs/reference/query-dsl/constant-score-query.asciidoc]: Good point - this really was more me being too lazy to reformat than anything else, but I agree it makes a lot of sense to keep these the same within one file.

File: docs/reference/query-dsl/geo-distance-query.asciidoc (hunk 2 above)
Comment (MaineC) [inline docs/reference/query-dsl/geo-distance-query.asciidoc]: I'll follow your suggestion. Wrt. "way too early" - essentially that's one of the reasons I put this PR out early, so we can look at things early as we progress and figure out what we like best.

Comment (nik9000) [inline docs/reference/query-dsl/geo-distance-query.asciidoc]: ++

File: docs/reference/query-dsl/bool-query.asciidoc [hunk 3]
```diff
@@ -81,7 +81,7 @@ all documents where the `status` field contains the term `active`.
 This first query assigns a score of `0` to all documents, as no scoring
 query has been specified:
 
-[source,json]
+[source,js]
```
Comment (nik9000) [inline docs/reference/query-dsl/bool-query.asciidoc]: Do you think we ought to complain for anything `[source,json]`? It'd be super easy to print a warning/fail the build.

Comment (nik9000) [inline docs/reference/query-dsl/bool-query.asciidoc]: I'd like us to be consistent here but I hesitate to pull the trigger on that.

File: docs/reference/query-dsl/constant-score-query.asciidoc [hunk 4]
```diff
@@ -7,12 +7,16 @@ filter. Maps to Lucene `ConstantScoreQuery`.
...
-        "filter" : {
-            "term" : { "user" : "kimchy"}
-        },
-        "boost" : 1.2
+    "query": {
+        "constant_score" : {
+            "filter" : {
+                "term" : { "user" : "kimchy"}
+            },
+                "boost" : 1.2
```
Comment (nik9000) [inline docs/reference/query-dsl/constant-score-query.asciidoc]: Extra indent I think.

**Comment by nik9000** [comment]
LGTM

File: docs/reference/query-dsl/bool-query.asciidoc (hunk 3 above)
Comment (MaineC) [inline docs/reference/query-dsl/bool-query.asciidoc]: No idea really, though in general I'm in favor of consistency - makes it easier for new comers to adopt existing style.

File: docs/reference/query-dsl/constant-score-query.asciidoc (hunk 4 above)
Comment (MaineC) [inline docs/reference/query-dsl/constant-score-query.asciidoc]: Thanks for catching. Now need to figure out what went wrong with vim auto-indent ;)

File: docs/reference/query-dsl/bool-query.asciidoc (hunk 3 above)
Comment (clintongormley) [inline docs/reference/query-dsl/bool-query.asciidoc]: In fact the syntax highlighter recognise `js`, not `json`, so yes would be good

**Comment by MaineC** [comment]
Modified the remaining query-dsl docs.

Before going ahead and merging I'd appreciate some feedback in particular to the changes made to docs/reference/query-dsl/geo-polygon-query.asciidoc and docs/reference/query-dsl/parent-id-query.asciidoc - examples in both needed a bit of adjustment to get past parsing at all. A second pair of eyes stating that the result is equal to the intent of the example before would be welcome.

File: docs/reference/query-dsl/bool-query.asciidoc (hunk 3 above)
Comment (nik9000) [inline docs/reference/query-dsl/bool-query.asciidoc]: I'll make a PR!

File: docs/reference/query-dsl/geo-polygon-query.asciidoc [hunk 5]
```diff
@@ -6,25 +6,29 @@ points. Here is an example:
...
-        "query" : {
-            "match_all" : {}
-        },
-        "filter" : {
-            "geo_polygon" : {
-                "person.location" : {
-                    "points" : [
+    "query": {
+        "bool" : {
+            "must" : {
```
Comment (nik9000) [inline docs/reference/query-dsl/geo-polygon-query.asciidoc]: Does it work if you leave that out?

Comment (nik9000) [inline docs/reference/query-dsl/geo-polygon-query.asciidoc]: Actually I wonder if it'd be better not to use it in the filter context? Like, can you just remove the bool entirely? If it needs the bool we should probably explain why it does. But that can be in another PR I think.

**Comment by nik9000** [comment]
I left a comment on the geo-polygon query but overall it looks pretty similar? I think the parent query changes are good.

File: docs/reference/query-dsl/geo-polygon-query.asciidoc (hunk 5 above)
Comment (MaineC) [inline docs/reference/query-dsl/geo-polygon-query.asciidoc]: I had the same thought when making those changes but decided to keep changes minimal and post-pone this to a later PR.

It does parse successfully without being surrounded by the bool query. Not sure if this is a left-over of the filter to query migration. @nknize @clintongormley

Comment (clintongormley) [inline docs/reference/query-dsl/geo-polygon-query.asciidoc]: It is only cacheable if it is inside the `bool.filter` clause, so I would leave it there.  (although perhaps delete the `bool.must.match_all`?

Comment (MaineC) [inline docs/reference/query-dsl/geo-polygon-query.asciidoc]: Makes sense. Will do in a separate PR and apply the same modification to any other reference/query-dsl/geo* docs that have the same structure.
